crates/motionfile/src/spline_interpolator.rs

```diff
@@ -83,20 +84,28 @@ impl InterpolatorError {
     }
 }
 
-impl TryFrom<MotionFile> for SplineInterpolator<Joints<f32>> {
+impl<T: Debug + Interpolate<f32>> TryFrom<MotionFile<T>> for SplineInterpolator<T> {
     type Error = InterpolatorError;
 
-    fn try_from(motion_file: MotionFile) -> Result<Self, InterpolatorError> {
+    fn try_from(motion_file: MotionFile<T>) -> Result<Self, InterpolatorError> {
         let mut current_time = Duration::ZERO;
         let mut keys = vec![Key::new(
             current_time,
             motion_file.initial_positions,
             Interpolation::Linear,
         )];
 
-        keys.extend(motion_file.frames.into_iter().map(|frame| {
-            current_time += frame.duration;
-            Key::new(current_time, frame.positions, Interpolation::Linear)
+        keys.extend(motion_file.frames.into_iter().map(|frame| match frame {
+            MotionFileFrame::Joints {
+                duration,
+                positions,
+            } => {
+                current_time += duration;
+                Key::new(current_time, positions, Interpolation::Linear)
+            }
+            MotionFileFrame::Condition(_) => {
+                panic!("Interpolator cannot be built with Conditions")
```

Comment:
Please propagate the error instead.

Reply:
This does not apply anymore, TryFrom MotionFile no longer possible